components/calendar/src/islamic.rs

```diff
@@ -582,6 +578,248 @@ impl UmmAlQura {
     }
 }
 
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+/// The inner date type used for representing [`Date`]s of [`IslamicCivil`]. See [`Date`] and [`IslamicCivil`] for more details.
+pub struct IslamicCivilDateInner(ArithmeticDate<IslamicCivil>);
+
+impl CalendarArithmetic for IslamicCivil {
+    fn month_days(year: i32, month: u8) -> u8 {
+        match month {
+            1 | 3 | 5 | 7 | 9 | 11 => 30,
+            2 | 4 | 6 | 8 | 10 => 29,
+            12 if Self::is_leap_year(year) => 30,
+            12 => 29,
+            _ => 0,
+        }
+    }
+
+    fn months_for_every_year(_year: i32) -> u8 {
+        12
+    }
+
+    fn days_in_provided_year(_year: i32) -> u32 {
+        355
```

Comment:
Leap years give an extra day in the year, so this should be a conditional based on whether it's a leap year

Reply:
done.